main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"unicode"
+)
+
+func IsSep(r rune) bool {
+	return unicode.IsSpace(r) || unicode.IsPunct(r) || unicode.IsSymbol(r)
+}
+
+func Words(s string) []string {
+	var w []string
+	var tmp []rune
+	wasSep := true
+	for _, r := range s {
+		if IsSep(r) {
+			if !wasSep {
+				w = append(w, string(tmp))
+				// tmp = tmp[:0]
+				tmp = nil
+			}
+			wasSep = true
+		} else {
+			wasSep = false
+			tmp = append(tmp, r)
+		}
+	}
+	if len(tmp) > 0 {
+		w = append(w, string(tmp))
+	}
+	return w
+}
+
+func find(word string, words []string) int {
+	for i, w := range words {
+		if w == word {
+			return i
+		}
+	}
+	return 0
```

Comment:
It's better to return `-1`, because otherwise the function `find` doesn't distinguish between a missing word and the first word (with index 0) in the input slice.

Reply:
Okay, I fixed it all, added more tests and committed. You can check whenever you can.

Reply:
Okay, I fixed it all, added more tests and committed. You can check whenever you can.